spec/namespaces/geospatial.yaml

```
@@ -0,0 +1,246 @@
+      responses:
+        '200':
+          $ref: '#/components/responses/geospatial.get_ip_to_geo_datasource@200'
+    put:
+      operationId: geospatial.put_ip_to_geo_datasource.0
+      x-operation-group: geospatial.put_ip_to_geo_datasource
+      x-version-added: '2.11'
+      description: |-
+        Create a specific IP2Geo data source.
+        Default values:
+          - `endpoint`: `"https://geoip.maps.opensearch.org/v1/geolite2-city/manifest.json"`
+          - `update_interval_in_days`: 3.
+      externalDocs:
+        url: 'https://docs.opensearch.org/docs/latest/ingest-pipelines/processors/ip2geo/#data-source-options'
+      parameters:
+        - $ref: '#/components/parameters/geospatial.put_ip_to_geo_datasource::path.datasource_name'
+      requestBody:
+        $ref: '#/components/requestBodies/geospatial.put_ip_to_geo_datasource'
+      responses:
+        '200':
+          $ref: '#/components/responses/geospatial.put_ip_to_geo_datasource@200'
+        '400':
+          $ref: '#/components/responses/geospatial.put_ip_to_geo_datasource@400'
+    delete:
+      operationId: geospatial.delete_ip_to_geo_datasource.0
+      x-operation-group: geospatial.delete_ip_to_geo_datasource
+      x-version-added: '2.11'
+      description: Delete a specific IP2Geo data source.
+      externalDocs:
+        url: 'https://docs.opensearch.org/docs/latest/ingest-pipelines/processors/ip2geo/#deleting-the-ip2geo-data-source'
+      parameters:
+        - $ref: '#/components/parameters/geospatial.delete_ip_to_geo_datasource::path.datasource_name'
+      responses:
+        '200':
+          $ref: '#/components/responses/geospatial.delete_ip_to_geo_datasource@200'
+        '404':
+          $ref: '#/components/responses/geospatial.delete_ip_to_geo_datasource@404'
+  /_plugins/geospatial/ip2geo/datasource/{datasource_name}/_settings:
+    put:
+      operationId: geospatial.put_ip_to_geo_datasource_settings.0
+      x-operation-group: geospatial.put_ip_to_geo_datasource_settings
+      x-version-added: '2.11'
+      description: Update a specific IP2Geo data source.
+      externalDocs:
+        url: 'https://docs.opensearch.org/docs/latest/ingest-pipelines/processors/ip2geo/#updating-an-ip2geo-data-source'
+      parameters:
+        - $ref: '#/components/parameters/geospatial.put_ip_to_geo_datasource_settings::path.datasource_name'
+      requestBody:
+        $ref: '#/components/requestBodies/geospatial.put_ip_to_geo_datasource_settings'
+      responses:
+        '200':
+          $ref: '#/components/responses/geospatial.put_ip_to_geo_datasource_settings@200'
+        '404':
+          $ref: '#/components/responses/geospatial.put_ip_to_geo_datasource_settings@404'
+
+components:
+  parameters:
+    geospatial.get_ip_to_geo_datasource::path.datasource_name:
+      name: datasource_name
+      in: path
+      required: true
+      schema:
+        type: string
```

Given this can be a comma delimited array, this should be:
```suggestion
        $ref: '../schemas/_common.yaml#/components/schemas/StringOrStringArray'
```